### DISCUSSION TITLE
github.githubassets.com times out

### DISCUSSION THREAD

**Discussion by ibn4n** [community/community#78053] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

General

### Body

I hope I'm posting this to the right place. Much of the github.com website is inaccessible from my home network (I'm posting this from my work network). Using the developer tools in Chrome, I was able to determine that connections to github.githubassets.com were timing out. A tracert to that ends as:

  6    18 ms    17 ms    17 ms  167.82.128.8
  7     *        *        *     Request timed out.
  8     *        *        *     Request timed out.

I looked up that last IP and it appears to be the Fastly CDN. I'm not sure where to go from here to troubleshoot further. 
![Untitled](https://github.com/community/community/assets/47467509/0080700f-7ef6-4ca1-bc55-a2c003a18a50)

**Comment by OshekharO** [comment]
Yes after GitHub started revamping it's ui, these problem started appearing i suggest you to use https://www.jsdelivr.com/github instead of GitHub cdn for these.

**Reply by hellorayza** [reply]
don't have issues tab 😞

**Comment by hellorayza** [comment]
When visiting GitHub pages, the page load time is extremely slow. After analyzing the HAR file, I found the following:

The bottleneck is github.githubassets.com (GitHub static assets CDN):

Average request time: 57 seconds

Breakdown: TTFB ≈ 25s, Download ≈ 32s

This indicates that the assigned CDN node is very slow, with poor throughput.

Other domains for comparison:

github.com: average ≈ 16s, mostly HTML, acceptable.

avatars.githubusercontent.com and fonts.gstatic.com: no significant issues.

Conclusion:

It’s not a local network or DNS issue, since tracert and ping are fine, and curl https://github.com/login completes in only 3 seconds.

The issue only occurs when loading static assets from github.githubassets.com.

Likely caused by suboptimal CDN node assignment or throttling.

**Comment by cjee21** [comment]
`github.githubassets.com` is so slow at certain times of the day that GitHub is almost unusable. Every page takes minutes to finish loading. It is sometimes as slow as 0.2Mbps and my connection to other sites is over 100Mbps during the same time.

The main GitHub homepage takes over 2 minutes to completely load when not yet cached by the browser:
226 requests
15.76 MB / 4.41 MB transferred
Finish: 2.29 min

Here is the tracert if it helps:
```
>tracert github.githubassets.com

Tracing route to github.githubassets.com [185.199.108.154]
over a maximum of 30 hops:

  1    <1 ms    <1 ms    <1 ms  ***
  2     3 ms     2 ms     5 ms  100.***.***.***
  3     2 ms     2 ms     5 ms  10.***.***.***
  4    17 ms    17 ms    16 ms  10.***.***.***
  5    71 ms    71 ms    71 ms  54113.sgw.equinix.com [27.111.228.38]
  6     *      103 ms   100 ms  cdn-185-199-108-154.github.com [185.199.108.154]

Trace complete.

```

My public facing IP is 175.143.*.* from AS4788.

`github.githubassets.com` does not even support IPv6 which may work better than IPv4 sometimes.

**Reply by cjee21** [reply]
`github.githubassets.com` supports IPv6 now but still the same issue. It's been months.

**Reply by olfek** [reply]
@cjee21 ---

Is `github.githubassets.com` slow for you too?

Most times I load a page, the layout is completely broken, presumably because CSS assets failed to load from the aforementioned host name.

**Reply by cjee21** [reply]
Very slow at certain times of the day then other times it's super fast. When it is the slowest, every page takes minutes to be functional. But only when using that fibre ISP. If I use my 5G mobile data then it is fast every time I use GitHub.

**Reply by olfek** [reply]
@cjee21 ---

What ISP?

**Reply by cjee21** [reply]
> What ISP?

AS4788 as mentioned above.

When it is slow, seems other sites using Fastly or Amazon CDN is slow too and images fail to load or load partially. So I think it is some routing issues between certain ISPs and CDNs. Maybe if GitHub can report to their CDN, it will be fixed more quickly since they are their CDN's customer.

**Reply by cjee21** [reply]
When it is slow, downloading from GitHub releases is slow too so it takes ages to obtain an app that should otherwise take seconds.

**Reply by ooshlablu** [reply]
This has been happening to me for the past few weeks also. The pages just get stuck on loading assets from github.githubassets.com and avatars.githubusercontent.com
I get pages that looks like mid-90's websites, and eventually the connections just time out and I never get a fully functional site. I might get one hours later. I've tried on multiple computers, and I have 2 ISPs, Starlink and Comcast, and the behaviour is the same on both.

**Reply by cjee21** [reply]
Seems fixed for me now. latency to github.githubassets.com via IPv6 is less than 50ms and all pages on GitHub loads completely in a second or two.

**Reply by c834606877** [reply]
in my side, the github.githubassets.com return ipv6 addr, but unable to access it. it have to force my client to use ipv4 as workaround

**Comment by mike-desjardins-anrok** [comment]
This has gotten horribly slow for me in recent days... I switched from google DNS to cloudflare - either it's helping, or I'm suffering from a placebo effect, I can't really tell.

**Reply by cjee21** [reply]
From experience, Cloudflare DNS makes many services worse because it does not support EDNS Client Subnet (ECS) and often routes to servers outside my country instead of to servers within my ISP.

**Reply by gabezal** [reply]
I'd say it's a placebo. This has nothing to do with the DNS provider. At least for me, Firefox can resolve github.githubassets.com just fine, as tested with about:networking#dnslookuptool, no matter what DNS settings I use (and no, cloduflare doesn't help either). It successfully resolves to both IPv6 and IPv4 addresses, so that's not the problem either.

Also, some github profile and project pages work fine. Only a small part seem to get stuck on github.githubassets.com. The current page and other project pages work fine, while github.com/labwc/ seems to be taking an infinite amount of time is still struggling to load from github.githubassets.com, for at least 10 minutes now.

I think this is a problem with Microsoft's (Azure??) servers. DuckDuckGo.com also uses Azure, and it also fails to load sometimes - rebooting my router fixes it. I investigated this a bit, and it turned out to be a MTU size reporting issue on Azure's part. I lowered the MTU size in my router to 1400, and that fixed it for a while, but whatever problem is plaguing Azure seems to be still there, and getting worse. Maybe this isn't really related to the githubassets issue, but due to the many issues I've had with Azure, that's where I'm pointing my finger.
